src/main/java/org/folio/rmapi/RMAPIService.java

```diff
@@ -97,6 +105,36 @@ public RMAPIService(String customerId, String apiKey, String baseURI, Vertx vert
     return this.getRequest(constructURL("vendors?search=zz12&offset=1&orderby=vendorname&count=1"), Object.class);
   }
 
+  public CompletableFuture<Vendors> retrieveProviders(String q, int page, int count, String sort) {
+    List<String> parameters = new ArrayList<>();
+    if (!Strings.isNullOrEmpty(q)) {
+      String encodedQuery;
+      try {
+        encodedQuery = URLEncoder.encode(q, "UTF-8");
+      } catch (UnsupportedEncodingException e) {
+        throw new IllegalStateException("failed to encode query using UTF-8");
+      }
+      parameters.add("search=" + encodedQuery);
+    }
+    parameters.add("offset=" + page);
+    parameters.add("count=" + count);
+    parameters.add("orderby=" + determineSortValue(sort, q));
+
+    return this.getRequest(constructURL("vendors?" + String.join("&", parameters)), Vendors.class);
+  }
```

Comment:
It may be helpful to have tests for RMAPIService layer itself?

Reply:
We can write tests for RMAPIService, but that will require a lot of mocking of vertx classes (HttpClient, HttpClientRequest etc.), if we need to test that correct request is constructed for RM API, then we can extract this logic and test it separately, like this - https://github.com/folio-org/mod-kb-ebsco-java/pull/25/commits/92b6e474e19ccdc79936782288df70428fbdb8a2